pages/edit-category/[id].js

```diff
@@ -11,107 +11,88 @@ import { FormMain, StyledForm, Input } from "../../components/FormComponents";
+  //STATES
+  const [categoryValidated, setCategoryValidated] = useState(true);
   const [submitButtonReady, setSubmitButtonReady] = useState(true);
-  //variable to check if the new category exists in the already existing categories
-  const [categoryExistsInCategories, setCategoryExistsInCategories] =
-    useState(false);
-  //variable to check if Modal Box for deletion is open
   const [deleteModalBoxOpen, setDeleteModalBoxOpen] = useState(false);
 
-  //HANDLING FUNCTIONS
-
-  function handleGoBack(event) {
-    event.preventDefault();
-    if (!pressedEnter) {
-      router.push(`/`);
-    } else {
-      setPressedEnter(false);
-    }
-  }
-
-  function handleKeyPress(event) {
-    switch (event.key) {
-      case "Enter": //check if user has not pressed enter on input field (for mobile check purposes "Go" or "Enter")
-        setPressedEnter(true);
-        break;
-      case "Backspace": //check if user has pressed backspace on input field (onChange won't work on backspace)
-        handleInput(event);
-        break;
-    }
-  }
+  const [currentInfo, setCurrentInfo] = useState(["", ""]);
 
+  //HANDLING FUNCTIONS
   function handleInput(event) {
-    checkInput();
-
-    function checkInput() {
-      let value = event.target.value;
-
-      if (!value.startsWith(" ") && value.length > 0) {
-        const inCategories = categoryInCategories(value);
-
-        setCategoryExistsInCategories(inCategories);
-        setSubmitButtonReady(!inCategories);
-      } else if (value.length > 0) {
-        value = value.trim();
-        event.target.value = value;
-        checkInput();
+    const value = event.target.value;
+    const inCategories = categoryInCategories(value.trim());
+
+    if (testHasChar.test(value)) {
+      if (!inCategories) {
+        setCategoryValidated(true);
+        setCurrentInfo(["", ""]);
+        setSubmitButtonReady(true);
       } else {
+        setCurrentInfo(["category", "Diese Kategorie existiert bereits"]);
+        setCategoryValidated(false);
         setSubmitButtonReady(false);
-        setCategoryExistsInCategories(false);
       }
+    } else {
+      setCategoryValidated(false);
+      setCurrentInfo([
+        "category",
+        "Eine Kategorie muss mind. ein Zeichen enthalten",
+      ]);
+      setSubmitButtonReady(false);
     }
   }
 
   function handleDelete() {
-    checkDelete();
-
-    function checkDelete() {
-      let standardCategory = categories.find(
-        (category) => category.default === true
-      );
-      let standardCategoryId = standardCategory ? standardCategory.id : null;
-
-      //no default category exist
-      if (!standardCategoryId) {
-        standardCategoryId = nanoid();
-        addCategory(standardCategoryId, "Standard Kategorie", true);
-      }
+    let standardCategory = categories.find(
+      (category) => category.default === true
+    );
+    let standardCategoryId = standardCategory ? standardCategory.id : null;
```

Comment:
why not using nanoId here instead of null? Then you can get rid of the let and the if statement. You can use !standardCategory && addCategory(...) afterwards

Reply:
@F-Kirchhoff i would only add a new standardCategory if no one is set 🤔 how would i do this without if-statement?